src/main/java/io/deephaven/book/PriceBook.java

```diff
@@ -706,7 +1100,50 @@ public static QueryTable build(@NotNull Table source,
                                    @NotNull String opColumnName,
                                    @NotNull String priceColumnName,
                                    @NotNull String... groupingCols) {
+
+        final PriceBook book = new PriceBook(source,
+                depth,
+                batchTimestamps,
+                timestampColumnName,
+                sizeColumnName,
+                sideColumnName,
+                opColumnName,
+                priceColumnName,
+                groupingCols);
+
+        return book.resultTable;
+    }
+
+    /**
+     * Build a book of bid and ask prices with the specified number of levels from the requested table, grouping input rows by the
+     * specified set of grouping columns.  Levels will be represented as a set of columns (Price, Time, Size) for each level.
+     *
+     * @param source the table with the source data
+     * @param snapshot the table with the book
+     * @param depth the desired book depth
+     * @param batchTimestamps set to true to batch input rows with identical timestamps into the a single output row.
+     * @param timestampColumnName the name of the source timestamp column
+     * @param sizeColumnName the name of the source size column
+     * @param sideColumnName the name of the source side column
+     * @param opColumnName the name of the source book-op column
+     * @param priceColumnName the name of the price column
+     * @param groupingCols the columns to group the source table by
+     *
+     * @return a new table representing the current state of the book.  This table will update as the source table updates.
+     */
+    @SuppressWarnings("unused")
+    public static QueryTable build(@NotNull Table source,
```

Comment:
There are enough arguments here to warrant switching to a Builder pattern.  We could use Immutables to generate the boilerplate builder itself and this method would simply take the result object.

Something like
```java
    public static QueryTable of(@NotNull final BookConfig config)
```

and used like this

```java
Pricebok.of(BookConfig.newBuilder()
    .thing1()
   .thing2()
   .build())
```

I realize that there are other factors at play so you might not have time to do this in this review,  but we should definitely do this as a follow on.